From eeb6ea586f95857405121be736e5ff58fc33bffb Mon Sep 17 00:00:00 2001
From: "Pillai, Venkatesh" <venkatesh.pillai@intel.com>
Date: Thu, 24 Aug 2023 09:48:04 +0530
Subject: [PATCH] Camera: Add aidl camera provider for external camera

As android.hardware.camera.provider@2.4::ICameraProvider is deprecated
in compatibility matrix for FCM Version 8, ICameraProvider AIDL packages
added.

Tracked-On: OAM-110079
Signed-off-by: Pillai, Venkatesh <venkatesh.pillai@intel.com>
---
 groups/camera-ext/ext-camera-only/product.mk       |  5 ++---
 groups/device-specific/caas/framework_manifest.xml | 11 +++--------
 groups/device-specific/caas/manifest.xml           | 12 +++---------
 3 files changed, 8 insertions(+), 20 deletions(-)

diff --git a/groups/camera-ext/ext-camera-only/product.mk b/groups/camera-ext/ext-camera-only/product.mk
index c715410..4902939 100644
--- a/groups/camera-ext/ext-camera-only/product.mk
+++ b/groups/camera-ext/ext-camera-only/product.mk
@@ -6,9 +6,8 @@ PRODUCT_COPY_FILES += \
 
 
 # External camera service
-PRODUCT_PACKAGES += android.hardware.camera.provider@2.4-external-service \
-                    android.hardware.camera.provider@2.4-service_64 \
-                    android.hardware.camera.provider@2.4-impl
+PRODUCT_PACKAGES += android.hardware.camera.provider-V1-external-impl \
+                    android.hardware.camera.provider-V1-external-service \
 #VHAL camera
 PRODUCT_PACKAGES += camera.$(TARGET_BOARD_PLATFORM) \
                     camera.$(TARGET_BOARD_PLATFORM).jpeg
diff --git a/groups/device-specific/caas/framework_manifest.xml b/groups/device-specific/caas/framework_manifest.xml
index fe4eb77..14d4d55 100644
--- a/groups/device-specific/caas/framework_manifest.xml
+++ b/groups/device-specific/caas/framework_manifest.xml
@@ -133,14 +133,9 @@
             <instance>default</instance>
         </interface>
     </hal>
-    <hal format="hidl">
-        <name>android.hardware.camera.provider</name>
-        <transport>hwbinder</transport>
-        <version>2.4</version>
-        <interface>
-            <name>ICameraProvider</name>
-            <instance>external/0</instance>
-        </interface>
+    <hal format="aidl">
+         <name>android.hardware.camera.provider</name>
+         <fqname>ICameraProvider/external/0</fqname>
     </hal>
     <hal format="hidl">
         <name>android.hardware.renderscript</name>
diff --git a/groups/device-specific/caas/manifest.xml b/groups/device-specific/caas/manifest.xml
index 8c47f24..096b1f5 100644
--- a/groups/device-specific/caas/manifest.xml
+++ b/groups/device-specific/caas/manifest.xml
@@ -59,15 +59,9 @@
         <fqname>@1.4::ICryptoFactory/clearkey</fqname>
         <fqname>@1.4::IDrmFactory/clearkey</fqname>
     </hal-->
-    <hal format="hidl">
-        <name>android.hardware.camera.provider</name>
-        <transport>hwbinder</transport>
-        <version>2.4</version>
-        <interface>
-            <name>ICameraProvider</name>
-            <instance>external/0</instance>
-            <instance>legacy/0</instance>
-        </interface>
+    <hal format="aidl">
+         <name>android.hardware.camera.provider</name>
+         <fqname>ICameraProvider/external/0</fqname>
     </hal>
     <hal format="hidl">
         <name>android.hardware.renderscript</name>
-- 
2.40.0

